core/sodasql/cli/cli.py

```diff
@@ -488,3 +498,143 @@ def scan(scan_yml_file: str, warehouse_yml_file: str, variables: tuple, time: st
+        warehouse_yml = warehouse_yml_parser.warehouse_yml
+
+        soda_server_client = create_soda_server_client(warehouse_yml)
+        if not soda_server_client.api_key_id or not soda_server_client.api_key_secret:
+            raise ValueError("Missing Soda cloud api key id and/or secret.")
+
+        with dbt_manifest.open("r") as manifest:
+            model_nodes, test_nodes = soda_dbt.parse_manifest(json.load(manifest))
+        with dbt_run_results.open("r") as run_results:
+            parsed_run_results = soda_dbt.parse_run_results(json.load(run_results))
+
+        models_with_tests = soda_dbt.create_models_to_tests_mapping(
+            model_nodes, test_nodes, parsed_run_results
+        )
+
+        tests_in_run_results = [
+            (run_result, test_nodes[run_result.unique_id])
+            for run_result in parsed_run_results
+            if run_result.unique_id in test_nodes.keys()
+        ]
+
+        tests = [
+            Test(
+                id=Parser.create_test_id(
+                    test_expression=test_node.compiled_sql if isinstance(test_node, CompiledSchemaTestNode) else None,
+                    test_name=test_run_results.unique_id,
+                    test_index=index,
+                    context_column_name=test_node.column_name,
+                    context_sql_metric_name=None,
+                    context_sql_metric_index=None,
+                ),
+                title=Parser.create_test_title(
+                    test_expression=test_node.compiled_sql if isinstance(test_node, CompiledSchemaTestNode) else None,
+                    test_name=test_run_results.unique_id,
+                    test_index=index,
+                    context_column_name=test_node.column_name,
+                    context_sql_metric_name=None,
+                    context_sql_metric_index=None,
+                ),
+                expression=test_node.compiled_sql if isinstance(test_node, CompiledSchemaTestNode) else None,
+                metrics=None,
+                column=test_node.column_name,
+            )
+            for index, (test_run_results, test_node) in enumerate(tests_in_run_results)
+        ]
+
+        test_results = {
+            test_run_results.unique_id:
+            TestResult(
+                test,
+                passed=test_run_results.status == TestStatus.Pass,
+                skipped=test_run_results.status == TestStatus.Skipped
+            )
+            for (test_run_results, _), test in zip(tests_in_run_results, tests)
+        }
+
+        for model_unique_id, test_unique_ids in models_with_tests.items():
+            start_scan_response = soda_server_client.scan_start(
+                warehouse_yml.name,
+                warehouse_yml.dialect.type,
+                table_name=model_nodes[model_unique_id].alias,
+                scan_yml_columns=None,
+                scan_time=datetime.now().isoformat(),
```

Comment:
the run results contain time per test, where we have a time per dataset. This also works, but is not _really_ the scan results: when the SQL is executed.